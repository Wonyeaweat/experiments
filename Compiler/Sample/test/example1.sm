program example1;
var tot,sum,len:integer;
    vis,flag,ok:bool;
    dp:integer;
    chr:char;
    this_is_a_very_long_variable_name:integer;
    A,B,C,X,Y:integer;
begin
  A := (3+2)*5;
  A := 3+2*5;
  C:=2*(7+8*9/10-5);
  A:=B*C+37;
  A:=42+B*C;
  X:=1;
  chr := this_is_a_very_long_variable_name;
  undefined := 233
end.
