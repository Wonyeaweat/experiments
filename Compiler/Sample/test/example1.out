( 0) (program, example1, - , - )
( 1) (+ , 3 , 2 , T1)
( 2) (* , T1, 5 , T2)
( 3) (:=, T2, - , A )
( 4) (* , 2 , 5 , T3)
( 5) (+ , 3 , T3, T4)
( 6) (:=, T4, - , A )
( 7) (* , 8 , 9 , T5)
( 8) (/ , T5, 10, T6)
( 9) (+ , 7 , T6, T7)
(10) (- , T7, 5 , T8)
(11) (* , 2 , T8, T9)
(12) (:=, T9, - , C )
(13) (* , B , C , T10)
(14) (+ , T10, 37, T11)
(15) (:=, T11, - , A )
(16) (* , B , C , T12)
(17) (+ , 42, T12, T13)
(18) (:=, T13, - , A )
(19) (:=, 1 , - , X )
(20) (:=, this_is_a_very_long_variable_name, - , chr)
(21) (:=, 233, - , undefined)
(22) (sys, - , - , - )
